### DISCUSSION TITLE
Palettes for plotnine

### DISCUSSION THREAD

**Discussion by JosephBARBIERDARNAL** [has2k1/plotnine#870] [General]
Hi!

I have a python library called [pypalettes](https://github.com/JosephBARBIERDARNAL/pypalettes) which is very similar to what the R package [paletteer](https://github.com/EmilHvitfeldt/paletteer) does but for Python (they are in fact almost all the same palettes).

I'd like to make pypalettes work with plotnine, but I'm not sure how to do it in practice. 

Are you interested? Would it be complicated to do? My library basically calls a csv and returns either a [LinearSegmentedColormap](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.LinearSegmentedColormap.html) or a [ListedColormap](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html) (both matplotlib objects) of the chosen palette.

**Comment by has2k1** [comment]
I like the variety and demo at the documentation site.

The core of the palettes is handled in [mizani](https://github.com/has2k1/mizani) and the API is private (`mizani._colors`) and still in flux and may be separated into another package.

In `mizani` I had to move away from matplotlib colormaps mainly to remove the dependency on matplotlib. Second to have colormaps whose interface is independent of how the colormaps are created, i.e. all colormaps have both continuous and discrete palettes whether it makes sense or not.

That said, for now `pypalettes` can be used by getting access to the component list of colours and passing that to plotnine or mizani to create the palettes that they know how to deal with. This isn't very user friendly and as there is duplication of work, it would be best to find common ground.

I'm will not have a lot of time in the next 2 months, but I'll look at `pypalettes` before I do any more significant work on `mizani._colors`. I also need fast colour transformations and have to think about all that as well.

**Reply by mschilli87** [reply]
@has2k1:Is there any update on this?

I am developing a tool that uses modules creating plots with matplotlib and I would like to create my own additional plots with `plotnine`.
However, my tool allows the user to customize the palette and as I rely on upstream for some plots, this customization is implemented in a way that I end up with a `matplotlib.ListedColormap`.
Is there any way to somehow convert that into a palette that I can use for my `plotnine` plots?
